cdap-ui/app/directives/log-viewer/log-viewer.js

```diff
@@ -24,6 +24,40 @@ function LogViewerController ($scope) {
     message: true
   };
 
+  this.hiddenColumns = {
+    time: false,
+    level: false,
+    source: false,
+    message: false
+  };
+
+  var theColumns = [];
+  var cols = this.configOptions;
+
+  if(cols['source']){
+    theColumns.push('source');
+  }
+  if(cols['level']){
+    theColumns.push('level');
+  }
+  if(cols['time']){
+    theColumns.push('time');
+  }
+
+  var collapseCount = 0;
+  this.collapseColumns = function() {
```

Comment:
Collapsing the columns when pressing the arrow button on the 'Messages' header. They collapse one after the other, from right to left